datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/resolvers/structuredproperties/UpsertStructuredPropertiesResolver.java

```diff
@@ -0,0 +1,171 @@
+package com.linkedin.datahub.graphql.resolvers.structuredproperties;
```

Comment:
The logic in this resolver seems like duplicate work to support replacing a structured property. This might be better using the [structured property patch builder](https://github.com/datahub-project/datahub/blob/master/entity-registry/src/main/java/com/linkedin/metadata/aspect/patch/builder/StructuredPropertiesPatchBuilder.java) do this work. Or the generic patch builder. The patch support for structured properties was built to add/remove/replace structured property values on entities. The read current value and mutate is inherent in the patch operation for many kinds of aspects.